src/theory/arith/arith_rewriter.cpp

```diff
@@ -56,6 +57,62 @@ bool evaluateRelation(Kind rel, const L& l, const R& r)
   }
 }
 
+/** Flatten the given node (with child nodes of the same kind) into a vector */
+void flatten(TNode t, std::vector<TNode>& children)
+{
+  Kind k = t.getKind();
+  for (const auto& child : t)
+  {
+    if (child.getKind() == k)
+    {
+      flatten(child, children);
+    }
+    else
+    {
+      children.emplace_back(child);
+    }
+  }
+}
+
+/**
+ * Flatten the given node (with child nodes of one of the given kinds) into a
+ * vector.
+ */
+[[maybe_unused]] void flatten(TNode t,
+                              Kind k1,
+                              Kind k2,
+                              std::vector<TNode>& children)
+{
+  Assert(t.getKind() == k1 || t.getKind() == k2);
+  for (const auto& child : t)
+  {
+    if (child.getKind() == k1 || child.getKind() == k2)
+    {
+      flatten(child, k1, k2, children);
+    }
+    else
+    {
+      children.emplace_back(child);
+    }
+  }
+}
+
+/** Flatten the given node (with child nodes of the same kind) */
+Node flatten(TNode t)
+{
+  Kind k = t.getKind();
+  bool canFlatten = std::any_of(
+      t.begin(), t.end(), [k](TNode child) { return child.getKind() == k; });
+  if (!canFlatten)
+  {
+    return t;
+  }
+  std::vector<TNode> children;
+  flatten(t, children);
+  Assert(children.size() >= 2);
+  return NodeManager::currentNM()->mkNode(t.getKind(), std::move(children));
```

Comment:
I don't think that the `std::move` is necessary here. `children` is passed by reference.

Reply:
Hm, indeed. We should consider adding an appropriate overload...